viewing_party/party.py

```diff
@@ -1,23 +1,154 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(movie_title, genre, rating):
+    if not movie_title or not genre or not rating:
+        return None
+    return {"title":movie_title, "genre":genre, "rating":rating}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+    
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+    
+    rating = 0.0
+    for movie in user_data["watched"]:
+        rating += movie["rating"]
+    
+    return rating / len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    most_watched_genre = ""
+    most_watched = 0
 
+    if not user_data["watched"]:
+        return None
+    
+    for movies in user_data["watched"]:
+        if 'genre' in movies:
+            genre = movies["genre"]
+            genre_count[genre] = genre_count.get(genre, 0) +1

```

Comment:
Nitpick: missing whitespace after `+`


```suggestion
            genre_count[genre] = genre_count.get(genre, 0) + 1
```